### ISSUE TITLE
Improvement of the documentation regarding "Sub States"

### ISSUE THREAD

**Issue by matulkum** [ngxs/store#566]
While I see the usefulness of splitting your state into smaller parts, I do not get what the actual gains are to define a "State Slice" as a State child of another. While it says here https://ngxs.gitbook.io/ngxs/advanced/sub-states :

> The relationship between these two are bound by their hierarchical order. 

it does not actually explains what it means in practice that "these two are bound". I can perfectly use them both without having them "bound". So there is no explanation what the benefit is to define one as a child of the other.

Maybe it is obvious for former Redux users? I never use it before.

**Comment by ddehghan** [comment]
I agree. Here are my questions. It would be great if someone can clarify this. 

How to I patch the state of a child state or parent state?
How do I dispatch an action within another action using data from parent or child state?

**Comment by ddehghan** [comment]
some good info from slack:
```
Jewsus [18 hours ago]
Or maybe you could @Action(ChildAction) from you parent state?

Mark Whitfeld [8 hours ago]
The advised way to go on this is to handle the action in the parent state too and make the necessary changes to the parent state there. An action class should not really be specific to the child or the parent because to is expressing a domain event
```
calling @Action(ChildAction)  from the parent works.

**Comment by clark-stevenson** [comment]
I am kind of in the same boat. 

I don't understand what `children` does once you get to that point. 

I honestly thought that if my childs state name was `child`,  and that it's parent was `children[ChildState]` that the unicorns would convert that into meaning that my parent could say `patchState({child: { visible: false}})`

But all answers that I can find, including this issue, refer to using the same Action on both parent and child but if that is the case then `children` doesn't do anything because I am effectively making 2 unrelated independent states.   I expect "ConfigureFooterSettings` to apply to my `FooterState` that in turn patches the visibility of its children.  That my child states actually don't have actions in this situation. 

I am probably totally off piste also almost certainly due to my lack of a NGRX background.

**Comment by ddehghan** [comment]
Also, can I place a child deep in the parent or it always goes to the root of the parent?

**Comment by splincode** [comment]
Released in 3.5.0
